payments_service/braintree/views.py

```diff
@@ -45,12 +45,37 @@ def replace_call_args(self, request, args, kw):
         The important part is that it only lets you get
         payment methods for the logged in user.
         """
+        if request.method.lower() != 'get':
+            return args, kw
+
         replaced_kw = {
             'active': kw.get('active', 1),  # active by default
             'braintree_buyer__buyer__uuid': request.user.uuid,
         }
         return tuple(), replaced_kw
 
+    def patch(self, request, pk=None):
+        """
+        Allow patching of payment methods.
+
+        * verify that the user wanting to make a patch is allowed too
+        * send through the patch
+        """
+        res = self.get(request, pk=pk)
+        # At the moment get returns a good or bad response as opposed to
+        # raising an error, but hides the status code from solitude.
+        #
+        # We'll assume anything not a 200 is 403.
+        if res.status_code != 200:
```

Comment:
I thought on a PATCH to an endpoint with a primary key (eg. `/braintree/mozilla/paymethod/1/`) they would be ignored. I just tried it out though and was surprised proved to be wrong, so yeah sounds like it should work and make this simpler.